### ISSUE TITLE
[Feature Request] "Unload tree" in the right click menu

### ISSUE THREAD

**Issue by KAMKillJoy** [mbnuqw/sidebery#1735]
### Description

It would be very convenient to have a menu item "Unload Group (Tree)" in the tab menu. Currently, I have to collapse the group and click on the "Unload" option to achieve the desired effect.

**Comment by JohnVKRDam** [comment]
You can already do this by right-clicking on the ICON of the parent Tab of the tree. So, do NOT right-click on the Tab name, but on the actual ICON of the parent Tab. Then select Unload, and the entire tree will be unloaded.

If this works could you please close this feature request? It will greatly help the developer from uncluttering his open feature requests.

**Comment by pe1uca** [comment]
Wow, this is very unintuitive.   
I think this feature request should be about better explaining this feature.    

It doesn't make sense the same option should have three different ways of doing two different actions:  
1. Click on tab name: unload given tab
2. Click on tab name of collapsed tree/group: unload tab and all children.  
3. Click on tab icon: unload tab and all children.    

This should be in two different options:  
- Unload: given tab.  
- Unload tree: given tab and children (should only be shown if the given tab is a group or a tree)

I think also the page of a group should have a dedicated "Unload all" button, since currently there's no clear indication of this feature, you only see the unload button of each tab in the group.

**Comment by JohnVKRDam** [comment]
@pe1uca 

I'm sorry, I respect your opinion, but I thoroughly disagree.

YOU don't like it, but others do. So, "it doesn't make sense" is your opinion, not a fact. It makes perfect sense to me and many others that have been using it for years now.

I most definitely DO NOT want these options to be changed. I absolutely LOVE the way I have multiple options to do operations on Tabs and Tab Trees, using a SINGLE action from the Context Menu!!

.

Developer, PLEASE do not do split this single action up into multiple! This would severely hamper the way a lot of us work on large Tab Sets! See my explanation right below.

.

The fact that you want to make those JUST THREE choices into different options in the Context Menu, will make my (and a LOT of people's) Context Menu MUCH LONGER, close to TWICE as long, thus LESS intuitive. That's because these combined right-click options are not just for unloading, but they are for more options that work on Tabs / Tab Trees. Such as:

- Load
- Unload
- Duplicate
- Copy Url(s)
- Copy Title(s)
- Set Color
- Move to Panel
and more...

As per your suggestion, this would then become:

- Load (Tab)
- Load (Tree)
- Unload (Tab)
- Unload (Tree)
- Duplicate (Tab)
- Duplicate (Tree)
- Copy Url (Tab)
- Copy Url (Tree)
- Copy Title (Tab)
- Copy Title (Tree)
- Set Color (Tab)
- Set Color (Tree)
- Move to Panel (Tab)
- Move to Panel (Tree)
and more...

Do you see how that doesn't really work?

And added to that, you want to make the Context Menu entirely unpredictable by having some options only show under certain conditions.... That is the most unintuitive thing EVER for a UI. Conditional display of options is a big NO-NO in UI development world.

.

Furthermore, the fact that these combined options have existed for a VERY long time already, and you didn't know about them, is kind of on you. Not everything HAS to be "intuitive", whatever that is, because THAT is very personal, as I DO find them intuitive. Some things you must LEARN, from a manual, or from asking, or from reading through the comments, incidents, and changes of this extension, or by experimenting.

.

Also, didn't you yourself just perfectly summarize what the options do? Was it really THAT hard to memorize??

- Right-click on Tab Name = Affects Tab Only
- Right-click on Tab Collapsed = Affects Entire Tree
- Right-click on Tab Icon = Affects Entire Tree

I'm sorry, but how hard can it be to memorize those three options... You could do it within an hour by just using them for a bit.

.

Sometimes you just need to RTFM or simply ask/read, and not think you should be able to know everything just by looking at it. Because frankly, that is IMPOSSIBLE.

There is no UI in the world that will learn you everything by just looking at it. That's just not how it works.

**Comment by pe1uca** [comment]
I just found out this is already partially implemented.  
When right-clicking on a single tab the menu shows "Copy URL" and "Copy title"  

![Image](https://github.com/user-attachments/assets/0099f239-ae09-4b2d-bcd2-d5b415cf45f4)  

When right-clicking on a collapsed tree or on the icon of the tree when is expanded it shows "Copy URLs" and "Copy titles"  

![Image](https://github.com/user-attachments/assets/3da66f0c-83c8-47b7-a629-0395f8e60ba3)  

Also, duplicating all options is implemented for the "Sort" sub menu, with the "Tree by *" options only being enabled when a tree was clicked (Regardless of the location of the click, i.e. name or icon)  

![Image](https://github.com/user-attachments/assets/d7e4b84a-ccdf-4308-98c8-708d7e9a5ef8)

---  

To keep consistency, and to improve the UI, all the options could properly describe the action will affect all tabs in the tree and only be shown when a tree has been clicked.  
(I still think clicking on the tab name vs the tab icon is very precise, but I can't think of any solution which also allows controlling the root tab)  

| Single tab | Tree |
| - | - |
| Move to | Move tree to |
|Reopen in | Reopen tree in|  
| Set color | Set tree's color |  
| Pin | Pin all tabs in tree |
| Duplicate | Duplicate tree |
| Unload | Unload tree |
| Copy URL | Copy tree's URLs |
| Copy title | Copy tree's titles |
| Clear cookies | Clear tree's cookies |
| Close | Close tree |
| Sync | Sync tree |
| Open in container... | Open tree in container... |

The rest of the options available to the tabs' menu seem to have the same behavior regardless of selecting a tab or a tree.

**Comment by the-nelsonator** [comment]
@pe1uca 

>When right-clicking on a single tab the menu shows "Copy URL" and "Copy title"

>When right-clicking on a collapsed tree or on the icon of the tree when is expanded it shows "Copy URLs" and "Copy titles"

Right click on root icon (which I didn't know about, thanks @JohnVKRDam!) or collapsed tree selects all tabs in the tree, that's why it shows plural. Same shows if you ctrl-leftclick or shift-leftclick multiple, not specific to tree right click

![Image](https://github.com/user-attachments/assets/3752c154-810d-42b4-a36e-8b5c5c59ae3e)

**Comment by pe1uca** [comment]
Yes @the-nelsonator, that's what I'm saying with "this is already partially implemented".  
Those ones are plural when selecting multiple tabs or in trees, but not all options which affect all tabs selected are plural or denote the actions will be executed for all tabs.

**Comment by the-nelsonator** [comment]
@pe1uca Gotcha! I'd argue it's fine this way ("Unload" doesn't imply singular but "Copy URL" does which is why that one needed pluralizing) but if you want every one to have "tab" or "tabs" depending on singular/plural, or to have separate labels for when a tree is selected (this would be new) you could open a new enhancement. The current logic is not specific to trees though, that's all I was saying

**Comment by JohnVKRDam** [comment]
> @the-nelsonator  I'd argue it's fine this way

I agree 👍 And you're welcome for the tip, of course 🙂👍

I respect everyone's opinion, obviously, but looking at the MASSIVE amount of questions, change requests, and incidents, I think the developer (who's sacrificing his personal time on this extension, at no cost) has much better things to do than working on what I think is a non-issue. It has been working just fine for years, and once you know, you know. Anyway, I think I made my own position clear, and that's about all the time I'm going to put into this 👍

**Comment by mbnuqw** [comment]
Closing as wontfix (has workaround). Couple of notes:
- Yes, current behavior with right-click on the tab's favicon (expand icon) is unclear and should be properly documented.
- I wouldn't like to add 'tree' (and 'trees' since we can select multiple trees) variants to context menu since it's a lot of i11n work, plus it would require additional logic to detect what was selected: tab/tabs/tree/trees (additional delay between right-click and showing context menu).
- What can be done:
  -  #2108
  - #2109
  - #2110
